### PR TITLE
Changed device component instantiation to be a separate method.

### PR DESCRIPTION
**Pull request by Kircheneer**
# Closes: #DNE
# What's Changed

`Device.save` now calls `Device.create_components` to create its components. The advantage lies in the fact that now that method can be used to populate the components when the `save` method is not available, such as with `bulk_create`.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Documentation Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by Kircheneer** [comment]
Is there a place where I should document the availability of this API to developers?

**Review by glennmatthews — COMMENTED** [review]
Definitely a step in the right direction, thank you!

As far as documentation goes, there isn't a great place at present, but I might suggest just adding a note in `nautobot/docs/models/dcim/device.md` for now?

**Comment by Kircheneer** [comment]
Also this would be a prime use case for `mkdocstrings` imo - but I found it wasn't set up properly. Do we have a working version of that somewhere where the `nautobot` package is available to the `mkdocs` container?

File: nautobot/docs/models/dcim/device.md [hunk 1]
```diff
@@ -15,3 +15,12 @@ Device names must be unique within a site, unless the device has been assigned t
...
 For Devices forming a group (Failover, Load-Sharing, Redundacy or similar) refer to [Device Redundancy Groups](deviceredundancygroup.md) model documentation.
+
+## Developer API
+
+The `Device` Django ORM class supports a method called `create_components`. This method is normally called during `Device.save`, which is called whenever you save create a device via the GUI or the REST API, but if you are working directly in the ORM and encounter one of the two following scenarios, `Device.save` is not used:
+
+- Usage of `Device.bulk_create`
+- Usage of `Device.save` during handling of the `nautobot_database_ready` signal (uses [historical models](https://docs.djangoproject.com/en/3.2/topics/migrations/#historical-models))
+
+In these cases you will have to 'manually' re-run `Device.create_components` in order to instantiate the [device type's](devicetype.md) component templates (interfaces, power ports, etc.).
```
Comment (jathanism) [inline nautobot/docs/models/dcim/device.md:26]:
```suggestion
The `Device` Django model class supports a method called `create_components()`. This method is normally called during `Device.save()`, which is called whenever you save create a Device via the GUI or the REST API, but if you are working directly in the ORM and encounter one of the two following scenarios, `Device.save()` is not called:

- Usage of `Device.objects.bulk_create()` to perform a bulk creation of Device objects
- Usage of `Device.save()` during handling of the `nautobot_database_ready` signal (which uses [historical models](https://docs.djangoproject.com/en/3.2/topics/migrations/#historical-models))

In these cases you will have to manually run `Device.create_components()` in order to instantiate the [device type's](devicetype.md) component templates (interfaces, power ports, etc.).
```

**Review by jathanism — CHANGES_REQUESTED** [review]

File: nautobot/docs/models/dcim/device.md [hunk 2]
```diff
@@ -15,3 +15,12 @@ Device names must be unique within a site, unless the device has been assigned t
...
 For Devices forming a group (Failover, Load-Sharing, Redundacy or similar) refer to [Device Redundancy Groups](deviceredundancygroup.md) model documentation.
+
+## Developer API
+
+The `Device` Django model class supports a method called `create_components()`. This method is normally called during `Device.save()`, which is called whenever you save create a Device via the GUI or the REST API, but if you are working directly in the ORM and encounter one of the two following scenarios, `Device.save()` is not called:
+
+- Usage of `Device.objects.bulk_create()` to perform a bulk creation of Device objects
+- Usage of `Device.save()` during handling of the `nautobot_database_ready` signal (which uses [historical models](https://docs.djangoproject.com/en/3.2/topics/migrations/#historical-models))
+
+In these cases you will have to manually run `Device.create_components()` in order to instantiate the [device type's](devicetype.md) component templates (interfaces, power ports, etc.).
```
Comment (bryanculver) [inline nautobot/docs/models/dcim/device.md:26]: My only objection to this is the documentation reads that this is a class function, not an instance function. Could you instead do something like `device_instance.create_components()`?

Comment (Kircheneer) [reply nautobot/docs/models/dcim/device.md]: Done!

**Comment by bryanculver** [comment]
@jathanism please re-review when you can

**Review by jathanism — APPROVED** [review]

**Pull request merged**